opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+import org.opensearch.sql.opensearch.storage.script.filter.lucene.LuceneQuery;
+
+/**
+ * Base class for query abstraction that builds a relevance query from function expression.
+ */
+public abstract class RelevanceQuery<T extends QueryBuilder> extends LuceneQuery {
+
+  protected Map<String, QueryBuilderStep<T>>
+      queryBuildActions;
+
+  public RelevanceQuery() {
+    super();
+    this.queryBuildActions = buildActionMap();
+  }
+
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    Iterator<Expression> iterator = func.getArguments().iterator();
```

Comment:
please add the check to see if the number of arguments is < 2

Reply:
that being said, is there a common place to extract the field and query arguments?  Seems like we're doing this in a lot of places...

Reply:
That common code is in the match_phrase PR. I will migrate that block of code from there to here.

Once our various implementations start to integrate, RelevanceQuery will be the base class for single field queries.

I expect there will be a similar one for multi-field queries.